consensus/core/src/base_committer.rs

```diff
@@ -227,17 +227,33 @@ impl BaseCommitter {
         leader_block: &VerifiedBlock,
         all_votes: &mut HashMap<BlockRef, bool>,
     ) -> bool {
+        let (gc_enabled, gc_round) = {
+            let dag_state = self.dag_state.read();   
+            (dag_state.gc_enabled(), dag_state.gc_round()) 
+        };
+
         let mut votes_stake_aggregator = StakeAggregator::<QuorumThreshold>::new();
         for reference in potential_certificate.ancestors() {
             let is_vote = if let Some(is_vote) = all_votes.get(reference) {
                 *is_vote
             } else {
                 let potential_vote = self
-                    .dag_state
-                    .read()
-                    .get_block(reference)
-                    .unwrap_or_else(|| panic!("Block not found in storage: {:?}", reference));
-                let is_vote = self.is_vote(&potential_vote, leader_block);
+                        .dag_state
+                        .read()
+                        .get_block(reference);
+                
+                let is_vote = if gc_enabled {
+                    if let Some(potential_vote) = potential_vote {
+                        self.is_vote(&potential_vote, leader_block)
+                    } else {
+                        assert!(reference.round <= gc_round, "Block not found in storage: {:?} , and is not below gc_round: {gc_round}", reference);
+                        false
+                    }
+                } else {
+                    let potential_vote = potential_vote.unwrap_or_else(|| panic!("Block not found in storage: {:?}", reference));
+                    self.is_vote(&potential_vote, leader_block)
+                };
+
```

Comment:
This is actually a fix not related to the DagState changes, but due to the gc feature in general. It's possible to have blocks that we check them as potential certificates, but with weak links that point to GC'ed blocks. In this case we don't want to fail/panic but instead detect that those ancestors are below GC and skip the checks